Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeUtil.java

```diff
@@ -137,54 +139,56 @@ private static String getEmailAddressAttrDisplayName() {
+                        } else {
+                            String sourceName = sourceArtifact != null ? "SourceArtifact display name: " + sourceArtifact.getDisplayName() : "SourceArtifact was null";
+                            logger.log(Level.WARNING, "Source artifact found through TSK_ASSOCIATED_ARTIFACT attribute was not a DataArtifact or "
+                                    + "an Analysis Result. AssociateArtifactAttr Value: {0} {1}",
+                                    new Object[]{assocArtifactAttr.getValueString(), sourceName});
+                        }
 
-                }
-            } else {
-                if (artifactTypeID == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
-                    BlackboardAttribute setNameAttr = analysisResult.getAttribute(BlackboardAttribute.Type.TSK_SET_NAME);
-                    if (setNameAttr != null && CorrelationAttributeUtil.getEmailAddressAttrDisplayName().equals(setNameAttr.getValueString())) {
-                        correlationAttrs.addAll(makeCorrAttrFromArtifactAttr(analysisResult, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD, CorrelationAttributeInstance.EMAIL_TYPE_ID, analysisResult.getAttributes()));
                     }
-                }
-                Content parent = analysisResult.getParent();
-                if (parent instanceof AbstractFile) {
-                    correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AbstractFile) parent));
-                } else if (parent instanceof AnalysisResult) {
-                    correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AnalysisResult) parent));
-                } else if (parent instanceof DataArtifact) {
-                    correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((DataArtifact) parent));
-                } else if (parent instanceof OsAccount) {
-                    for (OsAccountInstance osAccountInst : ((OsAccount) parent).getOsAccountInstances()) {
-                        if (osAccountInst.getDataSource().equals(analysisResult.getDataSource())) {
-                            correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch(osAccountInst));
-                            break;
+                } else {
+                    if (artifactTypeID == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
+                        BlackboardAttribute setNameAttr = analysisResult.getAttribute(BlackboardAttribute.Type.TSK_SET_NAME);
+                        if (setNameAttr != null && CorrelationAttributeUtil.getEmailAddressAttrDisplayName().equals(setNameAttr.getValueString())) {
+                            correlationAttrs.addAll(makeCorrAttrFromArtifactAttr(analysisResult, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD, CorrelationAttributeInstance.EMAIL_TYPE_ID, analysisResult.getAttributes()));
+                        }
+                    }
```

Comment:
Let's add a comment here:
"We no longer save email instances from keyword search hits in the central repository, but we still want to be able to search for email address instances in the CR when we are presenting email address keyword hits. Also note that we may want to correlate on the source Content  (parent) of the keyword hit as well, so we do not return at this point."